asserts/findwildcard.go

```diff
@@ -0,0 +1,94 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+/*
+findWildcard invokes foundCb for each regular file matching:
+
+<top>/<descendantWithWildcard[0]>/<descendantWithWildcard[1]...
+
+where each descendantWithWildcard component can contain the * wildcard;
+
+foundCb is invoked with the path of the file relative to top (that means top/
+ is excluded).
+
+Any I/O operation error stops the walking and bottoms out, so a foundCb invocation that returns an error.
```

Comment:
Looks like a missing word somewhere here, maybe: ", so *does* a foundCb invocation that returns an error."?